src/main/java/ch/heigvd/pro/b04/sessions/Session.java

```diff
@@ -4,40 +4,74 @@
 import java.sql.Timestamp;
 import java.util.Random;
 import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 import javax.persistence.CascadeType;
 import javax.persistence.Column;
 import javax.persistence.EmbeddedId;
 import javax.persistence.Entity;
 import javax.persistence.OneToMany;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.EqualsAndHashCode.Exclude;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 
 
-@Entity
 @NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Entity
+@EqualsAndHashCode
 public class Session {
+  private static final int CODE_BOUNDARY = 0xFFFF;
+
+  public enum SessionState {
+    OPEN, CLOSED_TO_NEW_ONES, CLOSED
+  }
```

Comment:
We might want to remove the `Session` prefix from `SessionState`, so we can us it by writing `Session.State`.